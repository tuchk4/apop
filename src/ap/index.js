import crop from './actions/splice';
import filter from './actions/filter';
import join from './actions/join';
import map from './actions/map';
import remove from './actions/remove';
import sort from './actions/sort';
import swap from './actions/swap';

import { apply, recursiveApply } from './utils/apply';

const sourceActions = { crop, filter, join, map, remove, sort, swap };

const ap = Object.assign(apply, sourceActions);
const recursive = Object.assign(recursiveApply, sourceActions);

export default Object.assign(ap, { recursive });
