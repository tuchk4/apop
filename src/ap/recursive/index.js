import crop from '../actions/splice';
import filter from '../actions/filter';
import join from '../actions/join';
import map from '../actions/map';
import remove from '../actions/remove';
import sort from '../actions/sort';
import swap from '../actions/swap';

import { recursiveApply } from '../utils/apply';

const sourceActions = { crop, filter, join, map, remove, sort, swap };
const recursiveActions = {};

Object.keys(sourceActions).forEach(action => {
  recursiveActions[action] = (...args) =>
    recursiveApply(sourceActions[action](...args));
});

const recursive = Object.assign(
  (...actions) => recursiveApply(...actions),
  recursiveActions
);

export default recursive;
