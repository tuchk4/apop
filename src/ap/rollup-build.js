// import { rollup } from 'rollup';
//
// import resolve from 'rollup-plugin-node-resolve';
// import uglify from 'rollup-plugin-uglify';
// import commonjs from 'rollup-plugin-commonjs';
// import babel from 'rollup-plugin-babel';
// import { minify } from 'uglify-es';
// import babelrc from 'babelrc-rollup';
//
// const babelConfig = {
//   presets: [['es2015', { loose: true, modules: false }, 'stage-0']],
//   plugins: ['external-helpers', 'transform-object-rest-spread'],
// };
//
// var rollUpConfig = {
//   format: 'umd',
//   sourceMap: 'inline',
//   plugins: [
//     resolve({
//       jsnext: true,
//       main: true,
//       browser: true,
//     }),
//     commonjs(),
//     babel(
//       babelrc({
//         addExternalHelpersPlugin: false,
//         config: babelConfig,
//         exclude: 'node_modules/**',
//       })
//     ),
//   ],
// };
//
// async function make(config) {
//   let bundle = await rollup({
//     entry: `./src/${config.moduleName}.js`,
//     plugins: config.plugins,
//   });
//
//   await bundle.write({
//     format: config.format,
//     moduleName: config.moduleName,
//     dest: `./${config.moduleName}.js`,
//     sourceMap: config.sourceMap,
//   });
//   await bundle.write({
//     format: config.format,
//     moduleName: config.moduleName,
//     dest: `./${config.moduleName}.min.js`,
//     sourceMap: config.sourceMap,
//     plugins: [...rollUpConfig.plugins, uglify({}, minify)],
//   });
//   console.log(`make ./${config.moduleName}.js and ./${config.moduleName}.min.js`);
// }
//
// (async function() {
//
//   const builds = ['rmkArray','recursive/rmkArray'];
//
//   builds.forEach(async moduleName => await make({...rollUpConfig, moduleName}));
//
// })();
