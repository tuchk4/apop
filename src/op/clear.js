import action from './actions/clear';
import { apply } from './utils/apply';

export default config => apply(action(config));
