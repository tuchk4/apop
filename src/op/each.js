import action from './actions/each';
import { apply } from './utils/apply';

export default config => apply(action(config));
