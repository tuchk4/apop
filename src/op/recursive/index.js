import actions from '../actions';

import { recursiveApply } from '../utils/apply';

const recursive = Object.assign((...args) => recursiveApply(...args), actions);

export default recursive;
