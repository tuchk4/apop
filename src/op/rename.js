import action from './actions/rename';
import { apply } from './utils/apply';

export default config => apply(action(config));
