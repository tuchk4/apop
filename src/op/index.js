import clear from './actions/clear';
import toSnakeCase from './actions/toSnakeCase';
import toCamelCase from './actions/toCamelCase';
import update from './actions/update';
import rename from './actions/rename';
import each from './actions/each';

import { apply, recursiveApply } from './utils/apply';

const sourceActions = { clear, toSnakeCase, toCamelCase, update, rename, each };
const hackActions = {};

Object.keys(sourceActions).forEach(key => {
  hackActions[key] = (...args) =>
    Object.assign(sourceActions[key](...args), { shortcut: true, key });
});

const op = Object.assign((...args) => apply(...args), hackActions);
const recursive = Object.assign(
  (...args) => recursiveApply(...args),
  hackActions
);

export default Object.assign(op, { recursive });
