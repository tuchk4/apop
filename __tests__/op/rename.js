import ap from '../../src/build/ap';
import op from '../../src/build/op';

const before = Object.seal({
  year: 2001,
});

test('Field function object', () => {
  const formula = op(
    op.rename({
      year: localState =>
        localState.year.toString().length === 4 ? 'fullYear' : 'year',
    })
  );

  const result = formula(before);
  expect(result).toMatchSnapshot();
});

test('Field static object', () => {
  const formula = op(
    op.rename({
      year: 'fullYear',
    })
  );

  const result = formula(before);
  expect(result).toMatchSnapshot();
});

test('Field function array', () => {
  const formula = ap(
    op.rename({
      year: localState =>
        localState.year.toString().length === 4 ? 'fullYear' : 'year',
    })
  );

  const result = formula([before]);
  expect(result).toMatchSnapshot();
});

test('Field static array', () => {
  const formula = ap(
    op.rename({
      year: 'fullYear',
    })
  );

  const result = formula([before]);
  expect(result).toMatchSnapshot();
});

// ---

test('Rename shorctu: Field function object', () => {
  const result = op.rename({
    year: localState =>
      localState.year.toString().length === 4 ? 'fullYear' : 'year',
  })(before);

  expect(result).toMatchSnapshot();
});

test('Rename shorctu: Field static object', () => {
  const result = op.rename({
    year: 'fullYear',
  })(before);

  expect(result).toMatchSnapshot();
});

test('Rename shorctu: Field function array', () => {
  const result = ap(
    op.rename({
      year: localState =>
        localState.year.toString().length === 4 ? 'fullYear' : 'year',
    })
  )([before]);

  expect(result).toMatchSnapshot();
});

test('Rename shorctu: Field static array', () => {
  const result = ap(
    op.rename({
      year: 'fullYear',
    })
  )([before]);

  expect(result).toMatchSnapshot();
});
