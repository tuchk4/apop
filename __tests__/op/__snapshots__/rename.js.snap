// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field function array 1`] = `
Array [
  Object {
    "fullYear": 2001,
  },
]
`;

exports[`Field function object 1`] = `
Object {
  "fullYear": 2001,
}
`;

exports[`Field static array 1`] = `
Array [
  Object {
    "fullYear": 2001,
  },
]
`;

exports[`Field static object 1`] = `
Object {
  "fullYear": 2001,
}
`;

exports[`Rename shorctu: Field function array 1`] = `
Array [
  Object {
    "fullYear": 2001,
  },
]
`;

exports[`Rename shorctu: Field function object 1`] = `
Object {
  "fullYear": 2001,
}
`;

exports[`Rename shorctu: Field static array 1`] = `
Array [
  Object {
    "fullYear": 2001,
  },
]
`;

exports[`Rename shorctu: Field static object 1`] = `
Object {
  "fullYear": 2001,
}
`;
